crates/superposition_types/src/custom_query.rs

```diff
@@ -0,0 +1,50 @@
+use std::collections::HashMap;
+
+use actix_web::web::Query;
+use regex::Regex;
+use serde::de::DeserializeOwned;
+use superposition_derives::CustomQuery;
+
+pub trait CustomQuery: Sized {
+    type Inner: DeserializeOwned;
+
+    fn regex_pattern() -> &'static str;
+    fn capture_group() -> &'static str;
+
+    fn query_regex() -> Regex {
+        Regex::new(Self::regex_pattern()).unwrap()
+    }
+
+    fn extract_query(
+        query_string: &str,
+    ) -> Result<Self, actix_web::error::QueryPayloadError> {
+        let query_map = Query::<HashMap<String, String>>::from_query(query_string)?;
+        let filtered_query = Self::filter_and_transform_query(query_map.into_inner());
+        let inner = Query::<Self::Inner>::from_query(&filtered_query)?.into_inner();
+        Ok(Self::from_inner(inner))
+    }
+
+    fn filter_and_transform_query(query_map: HashMap<String, String>) -> String {
+        let regex = Self::query_regex();
+        query_map
+            .into_iter()
+            .filter_map(|(k, v)| {
+                Self::extract_key(&regex, &k).map(|pk| format!("{pk}={v}"))
```

Comment:
just to verify, this is converting `platform[version]=version1` to `version=version1`, or am I missing here something ?

Reply:
Yup, that's what happening